components/locid/src/serde.rs

```diff
@@ -65,7 +65,12 @@ impl<'de> Deserialize<'de> for Language {
             }
         }
 
-        deserializer.deserialize_string(LanguageVisitor)
+        if deserializer.is_human_readable() {
+            deserializer.deserialize_string(LanguageVisitor)
+        } else {
+            Self::try_from_raw(Deserialize::deserialize(deserializer)?)
+                .map_err(serde::de::Error::custom)
+        }
```

Comment:
Praise: This is a good change.

Nit: Could be in a separate PR since it seems to be independent from the rest of the PR.

Reply:
It's very dependent, I'm adding this code